lib/reline.rb

```diff
@@ -265,31 +265,34 @@ def get_screen_size
 
     def readmultiline(prompt = '', add_hist = false, &confirm_multiline_termination)
       Reline.update_iogate
+      unless confirm_multiline_termination
+        raise ArgumentError.new('#readmultiline needs block to confirm multiline termination')
+      end
+
       io_gate.with_raw_input do
-        unless confirm_multiline_termination
-          raise ArgumentError.new('#readmultiline needs block to confirm multiline termination')
-        end
         inner_readline(prompt, add_hist, true, &confirm_multiline_termination)
+      end
 
-        whole_buffer = line_editor.whole_buffer.dup
-        whole_buffer.taint if RUBY_VERSION < '2.7'
-        if add_hist and whole_buffer and whole_buffer.chomp("\n").size > 0
-          Reline::HISTORY << whole_buffer
-        end
+      whole_buffer = line_editor.whole_buffer.dup
+      whole_buffer.taint if RUBY_VERSION < '2.7'
```

Comment:
Is the `taint` call necessary for Ruby 2.6? If not, I think it's fine to remove it now as well.

Reply:
`Kernel#gets` and other string read from IO are tainted in Ruby 2.6 https://docs.ruby-lang.org/en/2.6.0/Object.html#method-i-taint
Reline should return tainted string to align with `Kernel.gets`.
I'm not sure of the impact of removing taint. Perhaps nobody is using `tainted?`, but I'd like to keep this until dropping 2.6 support.
```ruby
# Code relying on taint security model
Foo::Settings.map do |s|
  if s.tainted?
    s=~/\A\d+\z/ ? s.to_i : s
  else
    eval s
  end
end
```